node_modules/oae-version/lib/api.js

```diff
@@ -79,20 +80,48 @@ var getVersion = module.exports.getVersion = function(callback) {
  * @api private
  */
 function _getHilaryVersion(callback) {
-    var hilaryDirectory = path.resolve(__dirname, '..', '..', '..');
     _getVersionInfo(hilaryDirectory, callback);
 }
 
 /**
- * Get the version information for the UI
+ * Get the 'version' for 3akai-ux, which mainly consists of the commit Hilary submodule is pointing to
  *
- * @param  {Function}   callback            Standard callback function
- * @param  {Object}     callback.err        An error that occurred, if any
- * @param  {String}     callback.version    The version information for the UI
+ * @param  {Function}   callback              Standard callback function
+ * @param  {Object}     callback.err          An error that occurred, if any
+ * @param  {String}     callback.version   The version information for the UI
  * @api private
  */
 function _getUIVersion(callback) {
-    _getVersionInfo(_uiPath, callback);
+
+    let submodulePointer = "";
+    let lastCheckedOutCommit = "";
+
+    gift.init(hilaryDirectory, (err, repo)  => {
+        repo.tree().contents((err, children) => {
+
+            // find the 3akai-ux submodule within the tree
+            const submodulePointer = _.find(children, (eachChild) => {
```

Comment:
You checked that this still works when 3akai is *not* a submodule? (For example when installing from package?)

Reply:
It won't work in any other situation other than the git submodule, because we're creating a reference CI/CD procedure which uses git-based installs. If the need for package installs arise in the future, we might have to re-think this approach, but hopefully that won't be necessary. Merging because all checks passed 🍾